src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -333,6 +333,9 @@ static bool TryRegisterNativeMembers (JniType nativeClass, Type marshalType, str
 
 			static bool FindAndCallRegisterMethod (Type marshalType, JniNativeMethodRegistrationArguments arguments)
 			{
+				if (!JniTypeManagerOptions.JniAddNativeMethodRegistrationAttributePresent)
```

Comment:
This would then need to become an instance method, and check `Runtime.JniAddNativeMethodRegistrationAttributePresent`.